Mapbox/MapboxMapsTests/Integration Tests/Map/MapViewIntegrationTests.swift

```diff
@@ -33,4 +33,32 @@ class MapViewIntegrationTests: IntegrationTestCase {
         }
         XCTAssertNil(weakMapView)
     }
+
+    func testUpdateFromDisplayLink() throws {
+        guard let rootView = rootViewController?.view else {
+            throw XCTSkip("No valid UIWindow or root view controller")
+        }
+
+        let resourceOptions = ResourceOptions(accessToken: accessToken)
+        let mapView = MapView(with: CGRect(origin: .zero, size: rootView.bounds.size), resourceOptions: resourceOptions, styleURL: .streets)
+        rootView.addSubview(mapView)
+
+        let originalFPS = mapView.preferredFPS
+
+        XCTAssertNotNil(mapView.displayLink)
+
+        mapView.preferredFPS = .lowPower
+
+        XCTAssertNotEqual(originalFPS, mapView.preferredFPS)
+        XCTAssertEqual(mapView.preferredFPS.rawValue, mapView.displayLink?.preferredFramesPerSecond)
+
+        mapView.displayLink = nil
+
+        mapView.preferredFPS = .maximum
+
+        // Test that the preferredFPS are not being updated when the displayLink is nil
+        XCTAssertNil(mapView.displayLink?.preferredFramesPerSecond)
+        XCTAssertNotEqual(mapView.preferredFPS.rawValue, mapView.displayLink?.preferredFramesPerSecond)
```

Comment:
Could we make this a separate test so that each test has a single focus?

Reply:
I split the tests up so that they are separate.